src/sas/sasgui/perspectives/fitting/fitpage.py

```diff
@@ -1600,91 +1615,129 @@ def _get_smear_info(self):
         Get the smear info from data.
 
         :return: self.smear_type, self.dq_l and self.dq_r,
-            respectively the type of the smear, dq_min and
-            dq_max for pinhole smear data
-            while dxl and dxw for slit smear
+            respectively the type of the smear, The average <dq/q> radial(p)
+            and <dq/q> theta (s)s for 2D pinhole resolution in % (slit is not
+            currently supported in 2D), (dq/q)_min and (dq/q)_max for 1D pinhole
+            smeared data, again in %, and dxl and/or dxw for slit smeared data
+            given in 1/A and assumed constant.
         """
-        # default
+        # set up defaults
         self.smear_type = None
         self.dq_l = None
         self.dq_r = None
         data = self.data
+        #sanity check - this should only be called if data exits
         if self.data is None:
             return
-        elif self.data.__class__.__name__ == "Data2D" or \
-            self.enable2D:
-            if data.dqx_data is None or data.dqy_data is None:
-                return
-            elif self.current_smearer is not None \
-                and data.dqx_data.any() != 0 \
-                and data.dqx_data.any() != 0:
+        #First check if data is 2D
+        #If so check that data set has smearing info and that none are zero.
+        #Otherwise no smearing can be applied using smear from data (a Gaussian
+        #width of zero will cause a divide by zero error)
+        if self.data.__class__.__name__ == "Data2D":
+            if data.dqx_data is not None and data.dqy_data is not None \
+              and data.dqx_data.all()and data.dqy_data.all():
                 self.smear_type = "Pinhole2d"
```

Comment:
The smearing type appears for 1D data, but not for 2D data. Should it appear at least in the text when "Use dQ data" is selected, just to be clear?

Reply:
Hadn't noticed and agree in principle.  However for 2D the original authors chose to replace the `type` with `accuracy`  on the grounds that the heavy computation here needed it I guess?  I suspect this was added much later at a user request and it was decided there really wasn't room for both? and/or that trying to keep both would be too ugly code wise?  I think this would be hard for space reasons.  Since 2D is not often used yet and ONLY allows pinhole I guess that is OK? perhaps at least OK enough to get us through to the new GUI of 5.0?  Or does anyone have a suggestions as to where/how to make the `pinhole2D` text appear?